move_base/include/move_base/move_base.h

```diff
@@ -86,30 +86,42 @@ namespace move_base {
 
       void clear()
       {
-        velocity = MIN_VELOCITY;
-        is_set = false;
+        dist_to_goal = 0.0;
+        average_velocity = MIN_VELOCITY;
+        time_to_goal = ros::Time(0, 0);
+        average_velocity_iteration = 0;
       }
 
-      double getTime(double distance, double velocity)
+      double calculateTimeToGoal(double distance, double velocity)
       {
         double epsilon = 0.01;
-        if(fabs(velocity - 0.0) < epsilon)
-        {
-          velocity = MIN_VELOCITY;
+        if (fabs(velocity - 0.0) < epsilon) {
+          return NAN;
+        }
+        else {
+          return fabs(distance/velocity);
         }
-
-        return fabs(distance/velocity);
       }
 
+      //it's on startup or local planner got a new plan
+      bool isNewPlan(double current_dist_to_goal)
+	  {
+        if ((time_to_goal == ros::Time(0,0)) || 
+          fabs(current_dist_to_goal - dist_to_goal) > 0.5) {
```

Comment:
I don't get this.

Reply:
The indication about the new plan, the robot has moved very fast during one iteration(0.1 s)

Reply:
You are right, it's very awkward, I will reset if I receive new_global_plan_.

Reply:
The function was removed.